internal/terraform/datadir/module_types.go

```diff
@@ -0,0 +1,36 @@
+package datadir
+
+import (
+	"github.com/hashicorp/go-getter"
+	tfregistry "github.com/hashicorp/terraform-registry-address"
+)
+
+type ModuleType string
+
+const (
+	NOTFOUND   ModuleType = "notfound"
+	TFREGISTRY ModuleType = "tfregistry"
+	LOCAL      ModuleType = "local"
+	GITHUB     ModuleType = "github"
+	GITLAB     ModuleType = "gitlab"
+)
+
+func (r *ModuleRecord) GetModuleType() ModuleType {
+	if _, err := tfregistry.ParseRawProviderSourceString(r.SourceAddr); err == nil {
+		return TFREGISTRY
+	}
+	if _, ok, _ := new(getter.GitHubDetector).Detect(r.SourceAddr, ""); ok {
+		return GITHUB
+	}
+	if _, ok, _ := new(getter.GitLabDetector).Detect(r.SourceAddr, ""); ok {
+		return GITLAB
+	}
```

Comment:
Terraform actually doesn't treat GitLab any differently - at least it doesn't use the GitLab detector at all:

https://github.com/hashicorp/terraform/blob/332ea1f233891e45bd274e714ab8e4e11c79dc95/internal/getmodules/getter.go#L30-L52

```suggestion
```